app/logic/Deployments.scala

```diff
@@ -33,18 +34,24 @@ object Deployments {
                        timestamp: OffsetDateTime,
                        links: Seq[Link],
                        note: Option[String],
-                       result: DeploymentResult): Kleisli[Future, Context, Deployment] =
+                       result: DeploymentResult): Kleisli[Future, Context, Deployment] = {
+
+    val deployment = Deployment(team, service, jiraComponent, buildId, timestamp, links, note, result)
+
     for {
-      deployment <- ES.Deployments
-        .create(team, service, jiraComponent, buildId, timestamp, links, note, result)
+      jiraResp     <- JIRA.createIssueIfPossible(deployment).local[Context](_.jiraCtx)
+      updatedLinks <- buildLinks(deployment, jiraResp).local[Context](_.jiraCtx)
+      identifiedDeployment <- ES.Deployments
+        .create(team, service, jiraComponent, buildId, timestamp, updatedLinks, note, result)
```

Comment:
Could you pass the `deployment` here, as you've already gone to the trouble of creating it? No reason to explode it out into fields again.